demo/resnet_1D.cpp

```diff
@@ -1165,8 +1165,17 @@ void inference(uint32_t const n_blocks = 2,
     //               << std::endl;
     // }
 
+// iterate over filters and free
+
 #if defined(NANO33BLE)
     small::detail::free_all();
+#else
+for (size_t l = 0; l < num_filters; l++)
+{
+    free(filter_buf_ptrs[l]);
```

Comment:
must use small::free_buffer(filter_buf_ptrs[l]); to ensure that delete is used when new is used in small::alloc_buffer() (address sanitizer flagged this too).